### PR TITLE
Full walkthrough

### PR DESCRIPTION
**Pull request by ErinBecker**
This PR completes the work started in https://github.com/datacarpentry/r-socialsci/pull/54. 

@fmichonneau - I need a bit of help with the formatting - it's not pulling in the data file, so none of the code chunks that rely on that step are running properly. Any advice?

### CODE REVIEW

File: _episodes_rmd/03-dplyr-tidyr.Rmd [hunk 1]
```diff
@@ -113,3 +113,596 @@ str(interviews)
...
+
+This format of the data allows us to do interesting things, like make a
+table showing the number of respondents in each village who
+owned a particular item:
+
+```{r, purl=FALSE}
+interviews_items_owned %>%
+filter(bicycle == TRUE) %>%
+group_by(village) %>%
+count(bicycle == TRUE)
```
Comment (fmichonneau) [inline _episodes_rmd/03-dplyr-tidyr.Rmd]: I think the last line should be: `count(bicycle)`

File: _episodes_rmd/03-dplyr-tidyr.Rmd [hunk 2]
```diff
@@ -113,3 +113,596 @@ str(interviews)
...
+```{r, purl = FALSE}
+interviews %>%
+group_by(village, memb_assoc) %>%
+summarize(mean_no_membrs = mean(no_membrs))
+```
+
+When grouping both by `village` and `membr_assoc`, we see rows in our
+table for respondents who did not specify whether they were a
+member of an irrigation association.
+that escaped before their sex could be determined and weighted.
```
[the remark this answers is not in the document]
Comment (ErinBecker) [reply _episodes_rmd/03-dplyr-tidyr.Rmd]: I dunno, that might make for a nice chuckle if we leave it in . . .

File: _episodes_rmd/03-dplyr-tidyr.Rmd [hunk 3]
```diff
@@ -113,3 +113,596 @@ str(interviews)
 ```
 
 Notice that the class of the data is now `tbl_df`
+
+This is referred to as a "tibble". Tibbles are data frames, but they tweak some of the old behaviors of data frames. The data structure is very similar to a data frame. For our purposes the only differences are that:
```
[the remark this answers is not in the document]
Comment (fmichonneau) [reply _episodes_rmd/03-dplyr-tidyr.Rmd]: :grimacing: it needs to be updated there as well then!

File: _episodes_rmd/03-dplyr-tidyr.Rmd [hunk 4]
```diff
@@ -113,3 +113,596 @@ str(interviews)
...
+mutate(people_per_room = no_membrs / rooms)
+head()
+```
+
+We may be interested in investigating whether being a member of an
+irrigation association had any effect on the ratio of household members
+to rooms. To look at this relationship, we will first remove
+data from our dataset where the respondent didn't answer the
+question of whether they were a member of an irrigation association.
+These cases are recorded as "NULL" in the dataset.
```
[the remark this answers is not in the document]
Comment (fmichonneau) [reply _episodes_rmd/03-dplyr-tidyr.Rmd]: alternatively for this, we could change it directly in the OpenRefine lesson.

File: _episodes_rmd/03-dplyr-tidyr.Rmd [hunk 5]
```diff
@@ -113,3 +113,596 @@ str(interviews)
...
+
+To recreate our original data frame, we will use the following:
+1. the data - `interviews_spread`
+2. the *key* column will be "respondent_wall_type" (as a character string). This is the name of the new column we want to create.
+3. the *value* column will be "dummy". This will be either TRUE or FALSE.
+4. the names of the columns we will use to fill the key variable are burntbricks:sunbricks (the column named "burntbricks" up to and including the column named "sunbricks").
+
+```{r, purl=FALSE}
+interviews_gather <- interviews_spread %>%
+gather(key = "respondent_wall_type", value = "dummy", burntbricks:sunbricks)
```
[the remark this answers is not in the document]
Comment (ErinBecker) [reply _episodes_rmd/03-dplyr-tidyr.Rmd]: We do in the 02-read episode:

# first element in the first column of the data frame (as a vector)
interviews[1, 1]
# first element in the 6th column (as a vector)
interviews[1, 6]
# first column of the data frame (as a vector)
interviews[, 1]
# first column of the data frame (as a data.frame)
interviews[1]
# first three elements in the 7th column (as a vector)
interviews[1:3, 7]
# the 3rd row of the data frame (as a data.frame)
interviews[3, ]
# equivalent to head_interviews <- head(interviews)
head_interviews <- interviews[1:6, ]`

`:` is a special function that creates numeric vectors of integers in increasing
or decreasing order, test `1:10` and `10:1` for instance.

File: _episodes_rmd/03-dplyr-tidyr.Rmd (hunk 4 above)
[the remark this answers is not in the document]
Comment (ErinBecker) [reply _episodes_rmd/03-dplyr-tidyr.Rmd]: @fmichonneau - good point. I didn't make these changes yet, but will put in another PR with them.

[the remark this answers is not in the document]
Comment (ErinBecker) [reply _episodes_rmd/03-dplyr-tidyr.Rmd]: It will require changing a few lines of code after the read_csv call.

**Pull request merged**